formats/gltf/read_test.go

```diff
@@ -1,7 +1,1430 @@
 package gltf_test
 
-import "testing"
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"image"
+	"image/color"
+	"image/png"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
 
-func TestRead(t *testing.T) {
+	"github.com/EliCDavis/polyform/formats/gltf"
+	"github.com/EliCDavis/polyform/modeling"
+	"github.com/EliCDavis/vector/vector2"
+	"github.com/EliCDavis/vector/vector3"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
 
+// =============================================================================
+// Test Data Generation Helpers
+// =============================================================================
+
+// intPtr returns a pointer to an integer
+func intPtr(i int) *int {
+	return &i
+}
+
+// float64Ptr returns a pointer to a float64
+func float64Ptr(f float64) *float64 {
+	return &f
+}
+
+// generateTestImage creates a simple test image with a solid color
+func generateTestImage(width, height int, col color.RGBA) image.Image {
+	img := image.NewRGBA(image.Rect(0, 0, width, height))
+	for y := 0; y < height; y++ {
+		for x := 0; x < width; x++ {
+			img.Set(x, y, col)
+		}
+	}
+	return img
+}
+
+// imageToBase64DataURI converts an image to a base64 data URI
+func imageToBase64DataURI(img image.Image) (string, error) {
+	buf := &bytes.Buffer{}
+	err := png.Encode(buf, img)
+	if err != nil {
+		return "", err
```

Comment:
In some tests, we're not checking if this function is returning an error, it might be better to just panic here if there's an error, or check.
```suggestion
		panic(err)
```